homeassistant/components/unifi/device_tracker.py

```diff
@@ -159,7 +160,7 @@ class UnifiTrackerEntityDescription(
         allowed_fn=async_client_allowed_fn,
         api_handler_fn=lambda api: api.clients,
         available_fn=lambda controller, obj_id: controller.available,
-        device_info_fn=lambda api, obj_id: None,
+        device_info_fn=async_client_device_info_fn,
```

Comment:
Morning! I've read through it once and I think the change to device tracker is wrong since Paulus changed that device trackers shouldn't create their own device entries.
This needs some more investigation. You could split this out to a separate PR and we can merge the other parts

Reply:
> I've read through it once and I think the change to device tracker is wrong since Paulus changed that device trackers shouldn't create their own device entries.

As I understand following the archi)tecture discussion, device trackers should be classified as secondary integration for device info.

Until then it is being linked using keys for primary integration, this PR fixes that.

More details at:
* [Documentation](https://developers.home-assistant.io/docs/device_registry_index/#categorizing-to-device-info)
* [Architecture discussion
](https://github.com/home-assistant/architecture/discussions/936)
* [PR implementation of device info categories](https://github.com/home-assistant/core/pull/95641)

> You could split this out to a separate PR and we can merge the other parts

I believe that the other changes (moving `async_client_device_info_fn` to `entity.py`)  can wait for the result of this.

Reply:
This is something device tracker specific. See previous PR changing this behaviour. I don't think it's actually documented properly.

https://github.com/home-assistant/core/pull/63328

Reply:
The logic of only adding device info / device tracker that already exists in the device registry is performed by:

https://github.com/home-assistant/core/blob/94870f05eee47d35d914c6f607b608c41be8e6c8/homeassistant/components/device_tracker/config_entry.py#L40-L74

This change does not impact this logic.

Anyway, I've changed it so that the device info for device tracker is categorized as a link instead of a secondary (since it's not expected to exist without a pre-existing device).

I point out that this change is necessary to correct the order of configuration entries in the device registry so that the frontend correctly displays device information.

I left the part of moving `async_client_device_info_fn` to `entity.py` to another PR.

Reply:
I don't see any other way to fix this other than this proposal, and I believe it's consistent with https://github.com/home-assistant/architecture/discussions/936.

I emphasize that this change does not change the logic that creates the device only if there is already a device with the same MAC address (implemented in #63328).

This PR just fixes the Unifi integration's categorization in the device registry, moving from primary integration to link. This change allows the frontend to use the correct icon/logo on the device page.

So if you can please review and let me know what can be done to remedy the issue https://github.com/home-assistant/frontend/issues/14620 (Unifi integration is the only known one that keeps this issue after implementations https://github.com/home-assistant/core/pull/95641 and https://github.com/home-assistant/frontend/pull/17105)

Reply:
You will have to get feedback on the architecture issue or hear by Paulus or someone else involved in the device tracker changes as I don't have the full picture of how they want to evolve devices.

_attr_device_info is expected to be only None.
https://github.com/home-assistant/core/blob/3624e30380f4664f9a8422def1536ed9fb723230/homeassistant/components/device_tracker/config_entry.py#L187

The device_info method is forced to None in the scanner class as well.
https://github.com/home-assistant/core/blob/3624e30380f4664f9a8422def1536ed9fb723230/homeassistant/components/device_tracker/config_entry.py#L322

Reply:
Unifi is supposed to be a secondary device info. A link entry is only meant to link up an entity if you're providing multiple entities for a single device. 

The frontend not acting properly is because although we differentiate now during entity registration, we don't actually store which config entry is the primary entry in the device, and so the frontend cannot use this information.

Reply:
Thank you for your attention @balloob!

> Unifi is supposed to be a secondary device info.

Hey @Kane610, please, could you look and see if you still need any information??

Reply:
I've talked with Paulus about this as I didn't think his answer satisfied my previous points and it still stands that the device tracker shouldn't directly define device entries so this PR is still not ok as it is right now. I suggest we close this. Maybe its better to talk in more general terms around this in an architectural discussion first.